cdc/log.cc

```diff
@@ -526,36 +551,173 @@ class transformer final {
                     r.for_each_cell([&](column_id id, const atomic_cell_or_collection& cell) {
                         auto& cdef = _schema->column_at(ckind, id);
                         auto* dst = _log_schema->get_column_definition(to_bytes("_" + cdef.name()));
-                        // TODO: collections.
-                        if (cdef.is_atomic()) {
-                            column_op op;
+                        auto res_type = static_pointer_cast<const tuple_type_impl>(dst->type)->type(1);
+                        column_op op = column_op::del;
+                        values[1] = values[2] = std::nullopt;
+                        auto has_pirow = pirow && pirow->has(cdef.name_as_text());
 
-                            values[1] = values[2] = std::nullopt;
+                        if (cdef.is_atomic()) {
                             auto view = cell.as_atomic_cell(cdef);
                             if (view.is_live()) {
                                 op = column_op::set;
                                 values[1] = view.value().linearize();
                                 if (view.is_live_and_has_ttl()) {
                                     values[2] = long_type->decompose(data_value(view.ttl().count()));
                                 }
-                            } else {
-                                op = column_op::del;
-                            }
+                            } 
+                        } else {
+                            auto mv = cell.as_collection_mutation();
+                            auto in = collection_mutation_input_stream(mv.data);
+                            auto tomb = tombstone{};
+                            auto now = gc_clock::time_point::min();
+
+                            std::optional<column_op> oop;
+                            op = column_op::add;
+                            std::vector<bytes> buf;
+
+                            auto process = [&](auto key_2_type, auto append_value) {
+                                auto has_tomb = in.read_trivial<bool>();
```

Comment:
This looks like an open-coded `collection_mutation_view::with_deserialized()`. Why not use that instead?

Reply:
The first iteration it was more different. It could perhaps be rewritten to use the existing helper now indeed.

Reply:
I think actually one reason was to avoid copying code between the caller lambdas instead, and also slightly less overhead, as we don't need all the views in a vector etc (we have other, funnier overhead vectors instead).

Reply:
Changed to use with_deserialized instead. Not sure it is much smaller, but reuses code so.